core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -806,6 +807,40 @@ private static String getFunctionType(FunctionMetadata function)
             };
         }
 
+        @Override
+        protected Node visitShowBranches(ShowBranches showBranches, Void context)
+        {
+            QualifiedObjectName tableName = createQualifiedObjectName(session, showBranches, showBranches.getTableName());
+            accessControl.checkCanShowBranches(session.toSecurityContext(), tableName);
+            getRequiredCatalogHandle(metadata, session, showBranches, tableName.catalogName());
+            if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.catalogName(), tableName.schemaName()))) {
+                throw semanticException(SCHEMA_NOT_FOUND, showBranches, "Schema '%s' does not exist", tableName.schemaName());
+            }
+            if (metadata.isMaterializedView(session, tableName)) {
+                throw semanticException(NOT_SUPPORTED, showBranches, "Relation '%s' is a materialized view, not a table", tableName);
+            }
+            if (metadata.isView(session, tableName)) {
+                throw semanticException(NOT_SUPPORTED, showBranches, "Relation '%s' is a view, not a table", tableName);
+            }
+            Optional<TableHandle> tableHandle = metadata.getRedirectionAwareTableHandle(session, tableName).tableHandle();
+            if (tableHandle.isEmpty()) {
+                throw semanticException(TABLE_NOT_FOUND, showBranches, "Table '%s' does not exist", tableName);
+            }
+
+            ImmutableList.Builder<Expression> rows = ImmutableList.builder();
+            for (String branch : metadata.listBranches(session, tableName)) {
+                rows.add(row(new StringLiteral(branch)));
+            }
```

Comment:
If there are no results, this will create an invalid rewrite:

```sql
SELECT branch_name Branch
FROM
   VALUES 
 branches (branch_name)
```

Reply:
Good catch! https://github.com/trinodb/trino/pull/26487 will fix the failure.